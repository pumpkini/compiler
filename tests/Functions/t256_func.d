int factorial(int n)
{
  if (n <=1 ) return 1;
  return n*factorial(n-1);
}

void main()
{
   int n;
   for (n = 1; n <= 15; n = n + 1)
      Print("Factorial(", n , ") = ", factorial(n), "\n");
}