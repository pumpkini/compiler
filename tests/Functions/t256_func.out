Factorial(1) = 1

Factorial(2) = 2

Factorial(3) = 6

Factorial(4) = 24

Factorial(5) = 120

Factorial(6) = 720

Factorial(7) = 5040

Factorial(8) = 40320

Factorial(9) = 362880

Factorial(10) = 3628800

Factorial(11) = 39916800

Factorial(12) = 479001600

Factorial(13) = 1932053504

Factorial(14) = 1278945280

Factorial(15) = 2004310016

